From f0d81b5ea1170274c57be7ee7989dc7bb0b64b84 Mon Sep 17 00:00:00 2001
From: Lee Duncan <lduncan@suse.com>
Date: Wed, 26 Jun 2013 11:57:39 -0700
Subject: Fix discovery error return without return value

openSUSE Build Service compiler noticed that one discovery
error path did not have a return value, so an appropriate
error is now being returned.

Signed-of-by: Lee Duncan <lduncan@suse.com>
---
 include/iscsi_err.h | 2 ++
 usr/iscsiadm.c      | 1 +
 2 files changed, 3 insertions(+)

diff --git a/include/iscsi_err.h b/include/iscsi_err.h
index 1139133..125f443 100644
--- a/include/iscsi_err.h
+++ b/include/iscsi_err.h
@@ -64,6 +64,8 @@ enum {
 	ISCSI_ERR_BUSY			= 28,
 	/* Operation failed, but retrying layer may succeed */
 	ISCSI_ERR_AGAIN			= 29,
+	/* unknown discovery type */
+	ISCSI_ERR_UNKNOWN_DISCOVERY_TYPE = 30,
 
 	/* Always last. Indicates end of error code space */
 	ISCSI_MAX_ERR_VAL,
diff --git a/usr/iscsiadm.c b/usr/iscsiadm.c
index 84d5a27..5030894 100644
--- a/usr/iscsiadm.c
+++ b/usr/iscsiadm.c
@@ -1235,6 +1235,7 @@ sw_discovery:
 		return do_isns(drec, ifaces, info_level, do_login, op);
 	default:
 		log_debug(1, "Unknown Discovery Type : %d\n", drec->type);
+		return ISCSI_ERR_UNKNOWN_DISCOVERY_TYPE;
 	}
 }
 
-- 
1.8.1.4

